encoding/csv.ts

```diff
@@ -0,0 +1,151 @@
+// Ported from Go:
+// https://github.com/golang/go/blob/go1.12.5/src/encoding/csv/
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+import { BufReader, BufState } from "../io/bufio.ts";
+import { TextProtoReader } from "../textproto/mod.ts";
+
+const INVALID_RUNE = ["\r", "\n", '"'];
+
+export class ParseError extends Error {
+  StartLine: number;
+  Line: number;
+  constructor(start: number, line: number, message: string) {
+    super(message);
+    this.StartLine = start;
+    this.Line = line;
+  }
+}
+
+export interface CsvParseOptions {
+  Comma: string;
+  Comment?: string;
+  TrimLeadingSpace: boolean;
+  LazyQuotes?: boolean;
+  FieldsPerRecord?: number;
```

Comment:
Fields should have a lower case first letter.

Reply:
Is there a way to add an eslint rule for this? I think it could be a good addition.